### DISCUSSION TITLE
optimal parameters for the subjects

### DISCUSSION THREAD

**Discussion by svumme** [TheLastBen/fast-stable-diffusion#992] [Q&A]
these days I tried to train the model with 55 images of a subject but I can't get a good result. The best test is this but not satisfactory:
I set UNet_Training_Step=6000
and Text_Encoder_Training_Steps=350
Text_Encoder_Concept_Training_Steps:0
and style_training not checked.
I also tried 12,000 steps and 20,000 but both went into overfitting.
Could you tell me how to calculate the various parameters based on a given number of images? I would be grateful to you.

**Comment by TheLastBen** [comment]
reduce the instance images to 15-20, too many instance images confuse the model, and set the training steps to 3000-4000

**Reply by svumme** [reply]
@TheLastBen  thank you, you are very kind. for the Text_Encoder_Training_Steps is considered a 10% of the total steps? so if I use UNet_Training_Step=4000 step,set the text encoder on 350-400? Is this the calculation that is done to get a good result? thank you

**Reply by TheLastBen** [reply]
the important thing is to keep Text_Encoder_Training_Steps below 500, you can push the concept_training steps up to 1000-1500, and the unet to 2000-3000

**Reply by svumme** [reply]
thank you!!!

**Reply by svumme** [reply]
hi, sorry to bother, i tried every possible setting of unet, 20000 , 12000, 9000, 6000, 4500, 4000, 3000, 1500 , 1200 steps.with these combinations of Text_Encoder_Training_Steps : 600, 450, 350.
and i tried with both sd 1.5 and 2.1 (512) . I worked with 89, 55, up to 15 photos. many tests but no good results. I also thought it was a photo problem, but I tried to use the ShivamShrirao dreambooth
and on the first try I got a good model. I think there is something not working right.

**Reply by TheLastBen** [reply]
what subject are you training ?

**Reply by svumme** [reply]
a 25 year old model, i'm a photographer and have several sets done both in the studio and outdoors. I don't know if it can be an indication on ShivamShrirao dreambooth... I used 1500 step, model 1.5, num_class_images=200 and I have an almost perfect model. I think there is a problem also because a friend of mine used your dreambooth some time ago and it works perfectly. I asked him to try again now and he also didn't get the previous result.
thanks

**Reply by TheLastBen** [reply]
what did you name the images ?

**Reply by svumme** [reply]
I searched for an impossible name : xtristtabab

**Reply by TheLastBen** [reply]
Try again with instance images that you can share, train the model, then send me the instance images so that I'll try from my side

**Reply by svumme** [reply]
i tried to do this thing. I removed all models from the automatic 1111 models folder and left only 1, I renamed the v2-1_512-nonema-pruned.yaml file with the same name as the model. But above all I noticed that putting the phrase "portrait photo of xtristtabab" at the beginning of the prompt now works! Is my procedure correct?

**Reply by TheLastBen** [reply]
So you're probably loading a different model that the one you trained, and also the V2 give awful results with faces, use the v1.5

**Reply by svumme** [reply]
probably it is as you say, there were over 100 models :)  i was reading on the checkpoint field of the webgui but maybe something was not working, because now it works fine. for 1.5 do I have to upload a yalm file? and which one is correct?
thanks

**Reply by TheLastBen** [reply]
I don't think you need to add a yaml for the v1.5

**Reply by svumme** [reply]
Thanks for all the replies, and sorry for the doubt I had. Now it works perfectly, the 1.5 is very good. I have a feeling the yalm works better. But it's just a feeling I have to do more tests.

**Reply by jjpb44** [reply]
Did you tried training with v2.1?
